engine/launcher/src/main/scala/org/enso/launcher/cli/InternalOpts.scala

```diff
@@ -80,38 +120,86 @@ object InternalOpts {
       )
       runDetachedAndExit(command)
     }
+
+    /**
+      * Tells the temporary launcher to remove the original launcher executable
+      * and possibly its (grand-)parent directory.
+      *
+      * @param executablePath path to the old executable
+      * @param parentToRemove path to the (grand-)parent directory; this
```

Comment:
The use of "(grand-)parent" is very confusing here.

Reply:
Hm, that's probably right.

What I tried to express is the fact that the `pathToRemove` may be the directory the binary is in (parent) or the parent directory of that directory.

This is because, someone may set `ENSO_BINARY_DIRECTORY` to `$ENSO_DATA_DIRECTORY` or `$ENSO_DATA_DIRECTORY/bin` and the idea was to try to handle these cases. This does not handle the case if someone sets `ENSO_BINARY_DIRECTORY` to `$ENSO_DATA_DIRECTORY/somethingelse` though. I guess I could do a similar trick with checking if the `ENSO_BINARY_DIRECTORY` is empty and removing it then.